core/src/main/java/de/danoeh/antennapod/core/service/download/DownloadService.java

```diff
@@ -190,10 +193,23 @@ public void onCreate() {
         cancelDownloadReceiverFilter.addAction(ACTION_CANCEL_ALL_DOWNLOADS);
         cancelDownloadReceiverFilter.addAction(ACTION_CANCEL_DOWNLOAD);
         registerReceiver(cancelDownloadReceiver, cancelDownloadReceiverFilter);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            this.registerLollipopNetworkListener();
+        }
 
         downloadCompletionThread.start();
     }
 
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    void registerLollipopNetworkListener() {
+        connectionMonitor.enable(getApplicationContext());
```

Comment:
It's just a single line of code, so I would just call it directly instead of wrapping it in a function.